src/Glpi/Asset/AssetDefinition.php

```diff
@@ -744,27 +750,46 @@ public function getFieldOrder(): array
      * @param bool $check_values
      * @return bool
      */
-    private function validateCapacityArray(mixed $capacities, bool $check_values = true): bool
+    private function validateCapacityArray(mixed &$capacities, bool $check_values = true): bool
     {
+        if ($capacities instanceof \stdClass) {
+            $capacities = (array)$capacities;
+        }
         if (!is_array($capacities)) {
             return false;
         }
 
         $is_valid = true;
+        $input_capacities = $capacities;
+        $capacities = [];
 
         $available_capacities = array_map(
             fn ($capacity) => $capacity::class,
             AssetDefinitionManager::getInstance()->getAvailableCapacities()
         );
-        foreach ($capacities as $classname) {
-            if (!is_string($classname)) {
+        //foreach ($capacities as $classname => $capacities_config) {
+        foreach ($input_capacities as $capacity) {
+            if ($capacity instanceof \stdClass) {
+                //on json_decode, we obtain a stdclass
+                $config = null;
+                if ($capacity->config) {
+                    $config = new CapacityConfig((array)$capacity->config);
+                }
+                $capacity = new Capacity(
+                    name: $capacity->name,
+                    config: $config
+                );
+            }
```

Comment:
This function was made to validate either the value in DB, either the value sent in `$input` by the form script. IMHO, this method should just validate the input/db value "format". The convertion to `Capacity` instances should be done in the `getDecodedCapacities()` method.

Reply:
It can indeed be done in `getDecodedCapacities()`; which must be used everytime (maybe not the case currently).